### PR TITLE
Profile/aw/56569/lighthouse dd api adapter

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- Works out a new way of calling the direct deposit api for updates and retrievals.
- Behavior is based on feature toggle `profile_lighthouse_direct_deposit`
- UI errors and overall layout remains the same, but behind the scenes plumbing changes results in:
  - new endpoint `/v0/profile/direct_deposits/disability_compensations` being used
  - redux store data for `vaProfile.cnpPaymentInformation` being flattened to hold data in consistent fashion between new and legacy api call results.
  - Error checking normalized to account for differences in response data shape.

The PR is pretty chonky and I apologize for having a lot to review. Most of the large line number changes revolve around mocks and organizational changes and not a huge amount of logic updates. I will comment throughout the diff notations to highlight main areas that I think are crucial for review.

For Platform COP Frontend team, the only real addition is adding the feature toggle name to the platform code. The toggle is already present in `vets-api` fyi.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/56569- *Link to ticket created in va.gov-team repo*

## Testing done

- Tests updated for new selector/action code and redux state shape updates.
- This is MVP and I plan on expanding on testing within ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/57407

## Screenshots
No visual differences.

## What areas of the site does it impact?
Profile - Direct Deposit

## Acceptance criteria

- [x] Lighthouse api is used when toggle is true.
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile/actions/paymentInformation.js [hunk 1]
```diff
@@ -43,30 +45,21 @@ export const EDU_PAYMENT_INFORMATION_SAVE_SUCCEEDED =
...
-
-  captureError(error, {
-    eventName: apiEventName,
-    code,
-    title,
-    detail,
-    status,
-  });
-};
-
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/actions/paymentInformation.js:63]: This code I found to be hard to follow and introducing unneeded complexity.

In addition, when I looked at errors in sentry that used this call, they almost always resulted in the code, title, detail and status being 'unknown' so it wasn't getting the right things and resulting in no value for us.

 Error responses are instead now being captured in a more straightforward way for profile via sentry.

File: src/applications/personalization/profile/actions/paymentInformation.js [hunk 2]
```diff
@@ -83,11 +76,21 @@ export function fetchCNPPaymentInformation(recordEvent = recordAnalyticsEvent) {
 
     if (response.error) {
       recordEvent({ event: `profile-get-cnp-direct-deposit-failed` });
+
+      captureCNPError(response, {
+        eventName: 'cnp-get-direct-deposit-failed',
+        useLighthouseDirectDepositEndpoint,
+      });
+
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/actions/paymentInformation.js:84]: This is the new capturing of error response and sending to sentry, just updates the event name, indicates if it was from lighthouse or legacy api, and then logs the error response, instead of trying to use a formatting function to pluck data from the error response.

File: src/applications/personalization/profile/mocks/server.js [hunk 3]
```diff
@@ -45,28 +46,45 @@ const responses = {
...
+    // successful update response
+    return res.status(200).json(paymentInformation.updates.success);
+  },
+  'GET /v0/profile/direct_deposits/disability_compensations': (_req, res) => {
+    // Lighthouse based API endpoint for direct deposit CNP
+    // alternate to the PPIU endpoint above: /v0/ppiu/payment_information
+    return res.json(disabilityComps.base);
+  },
+  'PUT /v0/profile/direct_deposits/disability_compensations': (_req, res) => {
+    return res.status(400).json(disabilityComps.errors.invalidRoutingNumber);
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/mocks/server.js:87]: You can change this to use `disabilityComps.updates.success` to simulate save success.

File: src/applications/personalization/profile/util/index.js [hunk 4]
```diff
@@ -36,16 +36,25 @@ export async function getData(apiRoute, options) {
...
+          message =>
+            message.key === errorKey &&
+            message.text?.toLowerCase().includes(errorText.toLowerCase()),
+        ),
+      ) ||
+      errors.some(
+        err =>
+          err?.code === errorKey &&
+          err?.detail?.toLowerCase().includes(errorText.toLowerCase()),
+      )
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/util/index.js:51]: This is new error checking for Lighthouse based response. The first section of error checking (before the || )can be removed once migration to new endpoint is complete.

File: src/applications/personalization/profile/util/direct-deposit.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    if (result.paymentAccount?.routingNumber) {
+      set(
+        result,
+        'paymentAccount.financialInstitutionRoutingNumber',
+        result.paymentAccount.routingNumber,
+      );
+    }
+    return result;
+  };
+}
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/util/direct-deposit.js:55]: So this is the new 'client adapter' class. It basically provides some convenience items to get the api endpoint to be used, format request options, and format response data.

I'd like to expand upon this and maybe move more Direct Deposit based code into this class, but it works for now.

Comment (Mottie) [reply src/applications/personalization/profile/util/direct-deposit.js:55]: I seem to recall from a Lighthouse meeting that they wanted to know who was using the ppiu endpoint from EVSS - I guess they know about this one?

Comment (adamwhitlock1) [reply src/applications/personalization/profile/util/direct-deposit.js:55]: Yep, they are aware, and basically this is in preparation of phasing that `ppiu` endpoint usage out to use a lighthouse based endpoint.

File: src/applications/personalization/profile/util/direct-deposit.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  constructor({ useLighthouseDirectDepositEndpoint = false }) {
+    this.useLighthouseEndpoint = useLighthouseDirectDepositEndpoint;
+  }
+
+  get endpoint() {
+    return this.useLighthouseEndpoint ? this.#LH_ENDPOINT : this.#PPIU_ENDPOINT;
+  }
+
+  generateApiRequestOptions(payload) {
```
Comment (allisonlu) [inline src/applications/personalization/profile/util/direct-deposit.js:17]: would it make sense to change `payload` to `fields` like you did in BankInfo.jsx? (just to stay consistent)

**Review by allisonlu — APPROVED** [review]
I left a comment on a variable name (non-blocking). everything else lgtm otherwise, no breaking changes on the UI.

**Review by rsmithadhoc — APPROVED** [review]
